semgrep-core/src/osemgrep/cli_scan/Core_runner.ml

```diff
@@ -184,27 +191,47 @@ let call_semgrep_core conf all_rules all_targets =
        else config
      in
   *)
+  (* TODO: we should not need to use Common.map below, because
+   * Run_semgrep.semgrep_with_raw_results_and_exn_handler can accept
+   * a list of targets with different languages! We just
+   * need to pass the right target object (and not a lang_job)
+   *)
   let lang_jobs = split_jobs_by_language all_rules all_targets in
   let results_by_language =
-    Common.map
-      (* !!!!Finally! this is where we branch to semgrep-core!!! *)
-      (Run_semgrep.semgrep_with_prepared_rules_and_targets config)
-      lang_jobs
+    lang_jobs
+    |> Common.map (fun lang_job ->
+           let _exn_optTODO, report, _filesTODO =
+             (* !!!!Finally! this is where we branch to semgrep-core!!! *)
+             Run_semgrep.semgrep_with_prepared_rules_and_targets config lang_job
+           in
+           report)
   in
-  merge_results results_by_language
+  let res = merge_results results_by_language in
+  (* TODO: should get this from Run_semgrep *)
+  let _exnTODO = None in
+  let files = [] in
+  (* similar to Run_semgrep.semgrep_with_rules_and_formatted_output *)
+  (* LATER: we want to avoid this intermediate data structure and
+   * return directly Out.cli_output
+   *)
+  let (match_results : Out.core_match_results) =
+    JSON_report.match_results_of_matches_and_errors files res
+  in
+  (* TOADAPT:
+      match exn with
+      | Some e -> Runner_exit.exit_semgrep (Unknown_exception e)
+      | None -> ())
+  *)
+  match_results
 
 (*************************************************************************)
 (* Entry point *)
 (*************************************************************************)
 
 (*
    Take in rules and targets and return object with findings.
-
-   ?core_opts_str: extra arguments passed to semgrep-core that need splitting
-   on whitespace. We will no longer be able to support this since
-   the semgrep-core CLI itself is going away.
 *)
-let invoke_semgrep (conf : Scan_CLI.conf) : result =
+let invoke_semgrep_core (conf : Scan_CLI.conf) : result =
```

Comment:
After the migration is done, I wonder if we'll still have a notion of core to represent the functionality of semgrep-core. If not (which is my current intuition), we'll have a single config object that derives from the merger between semgrep's and semgrep-core's config objects.

Reply:
yep, I hope so too. I'm trying to write the code with this in mind, so we can simply merge all the subdirs of osemgrep/ inside semgrep-core/src/ (and later do the big rename of semgrep-core).